diff --git a/http/Provision/Config/Nginx/server.tpl.php b/http/Provision/Config/Nginx/server.tpl.php
index 1158df5756ca13516f8eb16c34ba16a2742c95ed..4ce8cebf9583c17a5d335f2e691bba9e5531169c 100644
--- a/http/Provision/Config/Nginx/server.tpl.php
+++ b/http/Provision/Config/Nginx/server.tpl.php
@@ -16,7 +16,7 @@
   fastcgi_param  DOCUMENT_ROOT       $document_root;
   fastcgi_param  SERVER_PROTOCOL     $server_protocol;
   fastcgi_param  GATEWAY_INTERFACE   CGI/1.1;
-  fastcgi_param  SERVER_SOFTWARE     ApacheSolaris/$nginx_version;
+  fastcgi_param  SERVER_SOFTWARE     ApacheSolarisNginx/$nginx_version;
   fastcgi_param  REMOTE_ADDR         $remote_addr;
   fastcgi_param  REMOTE_PORT         $remote_port;
   fastcgi_param  SERVER_ADDR         $server_addr;
diff --git a/http/Provision/Service/http/fastcgi_params.conf b/http/Provision/Service/http/fastcgi_params.conf
index 70d62e38d15a93c5ce3bb8502664290c7efd95a7..e1991a74580e1b11f6942506f71de9929c308abe 100644
--- a/http/Provision/Service/http/fastcgi_params.conf
+++ b/http/Provision/Service/http/fastcgi_params.conf
@@ -12,7 +12,7 @@ fastcgi_param  DOCUMENT_URI       $document_uri;
 fastcgi_param  DOCUMENT_ROOT      $document_root;
 fastcgi_param  SERVER_PROTOCOL    $server_protocol;
 fastcgi_param  GATEWAY_INTERFACE  CGI/1.1;
-fastcgi_param  SERVER_SOFTWARE    ApacheSolaris/$nginx_version;
+fastcgi_param  SERVER_SOFTWARE    ApacheSolarisNginx/$nginx_version;
 fastcgi_param  REMOTE_ADDR        $remote_addr;
 fastcgi_param  REMOTE_PORT        $remote_port;
 fastcgi_param  SERVER_ADDR        $server_addr;
diff --git a/http/Provision/Service/http/fastcgi_ssl_params.conf b/http/Provision/Service/http/fastcgi_ssl_params.conf
index c24a765bcfe90216283d8177f9acead6bb749d46..2b7eedaf06eb12aa17017059c25f9a45555d6bcb 100644
--- a/http/Provision/Service/http/fastcgi_ssl_params.conf
+++ b/http/Provision/Service/http/fastcgi_ssl_params.conf
@@ -12,7 +12,7 @@ fastcgi_param  DOCUMENT_URI       $document_uri;
 fastcgi_param  DOCUMENT_ROOT      $document_root;
 fastcgi_param  SERVER_PROTOCOL    $server_protocol;
 fastcgi_param  GATEWAY_INTERFACE  CGI/1.1;
-fastcgi_param  SERVER_SOFTWARE    SSLApacheSolaris/$nginx_version;
+fastcgi_param  SERVER_SOFTWARE    ApacheSolarisSSLNginx/$nginx_version;
 fastcgi_param  REMOTE_ADDR        $remote_addr;
 fastcgi_param  REMOTE_PORT        $remote_port;
 fastcgi_param  SERVER_ADDR        $server_addr;
