diff -urp old/provision/platform/provision_drupal_settings.tpl.php new/provision/platform/provision_drupal_settings.tpl.php
--- old/provision/platform/provision_drupal_settings.tpl.php	2009-11-17 11:18:56.000000000 -0500
+++ new/provision/platform/provision_drupal_settings.tpl.php	2009-11-19 22:37:41.000000000 -0500
@@ -1,17 +1,12 @@
 
   $databases['default']['default'] = array(
-    'driver' => urldecode($_SERVER['db_type']),
-    'database' => urldecode($_SERVER['db_name']),
-    'username' => urldecode($_SERVER['db_user']),
-    'password' => urldecode($_SERVER['db_passwd']),
-    'host' => urldecode($_SERVER['db_host']),
+    'driver' => '<?php print $db_type; ?>',
+    'database' => '<?php print $db_name; ?>',
+    'username' => '<?php print $db_user; ?>',
+    'password' => '<?php print $db_passwd; ?>',
+    'host' => '<?php print $db_host ?>',
   );
-  $db_url = "<?php print strtr("%db_type://%db_user:%db_passwd@%db_host/%db_name", array(
-    '%db_type' => '$_SERVER[db_type]',
-    '%db_user' => '$_SERVER[db_user]', 
-    '%db_passwd' => '$_SERVER[db_passwd]',
-    '%db_host' => '$_SERVER[db_host]', 
-    '%db_name' => '$_SERVER[db_name]')); ?>";
+  $db_url = '<?php print strtr("%db_type://%db_user:%db_passwd@%db_host/%db_name", array('%db_type' => urlencode($db_type), '%db_user' => urlencode($db_user), '%db_passwd' => urlencode($db_passwd), '%db_host' => urlencode($db_host), '%db_name' => urlencode($db_name))); ?>';
 
   $profile = "<?php print $profile ?>";
   /**
@@ -48,6 +43,9 @@
   $conf['file_downloads'] = 1;
   $conf['cache'] = 1;
   $conf['clean_url'] = 1;
+  $conf['preprocess_css'] = 1;
+  $conf['preprocess_js'] = 1;
+  $conf['page_compression'] = 0;
 
   <?php if ($site_offline) : ?>
     $conf['site_offline'] = 1;
diff -urp old/provision/web_server/provision_apache.drush.inc new/provision/web_server/provision_apache.drush.inc
--- old/provision/web_server/provision_apache.drush.inc	2009-10-15 06:37:38.000000000 -0400
+++ new/provision/web_server/provision_apache.drush.inc	2009-11-19 22:35:33.000000000 -0500
@@ -213,6 +213,13 @@ function _provision_apache_create_config
     $text = provision_render_config($template, $data);
     fwrite($file, $text);
     fclose($file);
+    // Add below 6 lines to create symlink for nginx wildcard virtual domains
+    $nginx_publish_path = drush_get_option('publish_path');
+    $nginx_wildcard = '/data/u/';
+    $nginx_site_url = drush_get_option('site_url');
+    $nginx_site_url = $nginx_wildcard . $nginx_site_url;
+    if(is_link($nginx_site_url)) { unlink($nginx_site_url); }
+    symlink($nginx_publish_path,$nginx_site_url);
   }
 }
 
