diff -urp a/apps.profile.inc b/apps.profile.inc
--- a/apps.profile.inc	2012-10-14 06:42:00.000000000 -0400
+++ b/apps.profile.inc	2012-10-14 06:42:48.000000000 -0400
@@ -177,7 +177,7 @@ function apps_profile_download_batch_fin
       'title' => t('Downloading updates failed:'),
       'items' => $results['errors'],
     );
-    drupal_set_message(theme('item_list', $error_list), 'error');
+    //drupal_set_message(theme('item_list', $error_list), 'error');
   }
   elseif ($success) {
     drupal_set_message(t('Updates downloaded successfully.'));
