diff --git a/commerce_kickstart.install b/commerce_kickstart.install
index 238f3e6..37f55f8 100644
--- a/commerce_kickstart.install
+++ b/commerce_kickstart.install
@@ -203,7 +203,7 @@ function commerce_kickstart_configure_store_form() {
     '#title' => st('Do you want to install the demo store?'),
     '#description' => st('Shows you everything Commerce Kickstart can do. Includes a custom theme, sample content and products.'),
     '#options' => $options,
-    '#default_value' => '1',
+    '#default_value' => '0',
   );
   $form['localization'] = array(
     '#type' => 'fieldset',
