Index: drush_make.utilities.inc
===================================================================
RCS file: /cvs/drupal-contrib/contributions/modules/drush_make/Attic/drush_make.utilities.inc,v
retrieving revision 1.1.2.55
diff -u -p -r1.1.2.55 drush_make.utilities.inc
--- drush_make.utilities.inc	12 Nov 2010 08:19:52 -0000	1.1.2.55
+++ drush_make.utilities.inc	23 Nov 2010 13:02:49 -0000
@@ -454,7 +454,7 @@ function drush_make_get_data($data_sourc
   }
   // Remote file.
   else {
-    $file = _drush_make_download_file(array('url' => $datasource));
+    $file = _drush_make_download_file(array('url' => $data_source));
     $data = file_get_contents($file);
     drush_op('unlink', $file);
   }
