diff -urp a/openacademy/openacademy.info b/openacademy/openacademy.info
--- a/openacademy/openacademy.info	2012-05-31 14:51:19.000000000 -0400
+++ b/openacademy/openacademy.info	2012-06-03 12:35:45.000000000 -0400
@@ -31,6 +31,8 @@ dependencies[] = views
 dependencies[] = views_content
 dependencies[] = views_ui
 dependencies[] = token
+dependencies[] = ds
+dependencies[] = ds_extras
 
 ; Panopoly - Contrib - Field UI and Content Types
 dependencies[] = tablefield
@@ -70,13 +72,6 @@ dependencies[] = simplified_menu_admin
 dependencies[] = references_dialog 
 dependencies[] = backports
 
-; Panopoly - Contrib - Search 
-dependencies[] = search_api
-dependencies[] = search_api_solr
-dependencies[] = facetapi
-dependencies[] = search_api_facetapi
-dependencies[] = search_api_views
-
 ; Panopoly - Contrib - Products
 dependencies[] = apps 
 dependencies[] = features
@@ -86,9 +81,6 @@ dependencies[] = defaultcontent
 dependencies[] = strongarm
 dependencies[] = libraries
 
-; Panopoly - Contrib - Performance
-dependencies[] = redis
-
 ; Panopoly - Contrib - Development
 dependencies[] = devel
 dependencies[] = devel_generate
diff -urp a/openacademy/openacademy.profile b/openacademy/openacademy.profile
--- a/openacademy/openacademy.profile	2012-05-29 18:39:27.000000000 -0400
+++ b/openacademy/openacademy.profile	2012-06-03 12:38:48.000000000 -0400
@@ -26,7 +26,6 @@ function openacademy_install_tasks($inst
       'panopoly_images',
       'panopoly_magic',
       'panopoly_pages',
-      'panopoly_search',
       'panopoly_theme',
       'panopoly_users',
       'panopoly_widgets',
@@ -38,7 +37,6 @@ function openacademy_install_tasks($inst
       'panopoly_images',      
       'panopoly_magic',             
       'panopoly_pages',                   
-      'panopoly_search',                        
       'panopoly_theme',                               
       'panopoly_users',                                     
       'panopoly_widgets',                                         
@@ -315,7 +313,12 @@ function openacademy_theme_form($form, &
 function openacademy_theme_form_submit($form, &$form_state) {
   
   // Enable and set the theme of choice
-  $theme = $form_state['input']['theme'];
+  if (defined('DRUSH_BASE_PATH')) {
+    $theme = "openacademy_default";
+  }
+  else {
+    $theme = $form_state['input']['theme'];
+  }
   theme_enable(array($theme));
   variable_set('theme_default', $theme);
  
