diff --git a/includes/file.inc b/includes/file.inc
index c5e5cf07d636d2454c3e158eda0e2c0f6f7297fa..57a4e4734a8a175c4ae8dd8894a281699529b95b 100644
--- a/includes/file.inc
+++ b/includes/file.inc
@@ -476,6 +476,9 @@ function file_ensure_htaccess() {
  *   The default is TRUE which indicates a private and protected directory.
  */
 function file_create_htaccess($directory, $private = TRUE) {
+  // Skip this on BOA since .htaccess is never used in Nginx.
+  return;
+
   if (file_uri_scheme($directory)) {
     $directory = file_stream_wrapper_uri_normalize($directory);
   }
