diff --git a/Provision/Config/cdn.tpl.php b/Provision/Config/cdn.tpl.php
index a4ce098ea323d9855642ab4b17a54d93cb2d1d0f..00597f8795768bc43dea4fa0eae8afca75a22f67 100644
--- a/Provision/Config/cdn.tpl.php
+++ b/Provision/Config/cdn.tpl.php
@@ -3,7 +3,16 @@ $ip_address = !empty($ip_address) ? $ip_address : '*';
 ?>
 server {
   limit_conn   gulag 18; # like mod_evasive - this allows max 18 simultaneous connections from one IP address
-  listen       <?php print $ip_address . ':' . $http_port; ?>;
+<?php
+if ($ip_address == '*') {
+  print "  listen       {$ip_address}:{$http_port};\n";
+}
+else {
+  foreach ($server->ip_addresses as $ip) {
+    print "  listen       {$ip}:{$http_port};\n";
+  }
+}
+?>
   server_name <?php foreach ($this->cdn as $cdn_domain) : if (trim($cdn_domain)) : ?> <?php print $cdn_domain; ?><?php endif; endforeach; ?>;
   root         <?php print "{$this->root}"; ?>;

diff --git a/Provision/Config/cdn_disabled.tpl.php b/Provision/Config/cdn_disabled.tpl.php
index 396e309a3e746794759bb910edeb500bea38aa41..cee2eb0acaabb7304cd4cc9e61f5d81d4a1fd3bf 100644
--- a/Provision/Config/cdn_disabled.tpl.php
+++ b/Provision/Config/cdn_disabled.tpl.php
@@ -2,10 +2,19 @@
 $ip_address = !empty($ip_address) ? $ip_address : '*';
 ?>
 server {
-  listen       <?php print $ip_address . ':' . $http_port; ?>;
+  limit_conn   gulag 18;
+<?php
+if ($ip_address == '*') {
+  print "  listen       {$ip_address}:{$http_port};\n";
+}
+else {
+  foreach ($server->ip_addresses as $ip) {
+    print "  listen       {$ip}:{$http_port};\n";
+  }
+}
+?>
   server_name  <?php print implode(' ', $this->cdn); ?>;
   root         /var/www/nginx-default;
   index        index.html index.htm;
-
   ### Do not reveal Aegir front-end URL here.
 }
