diff -urp purge-one/purge.inc purge/purge.inc
--- a/purge.inc 2012-02-04 01:16:14.000000000 +0000
+++ b/purge.inc 2012-02-04 01:21:35.000000000 +0000
@@ -119,8 +119,15 @@ function purge_issue_requests_curl($purg
   // Execute the purge requests
   ob_start();
   do {
-    curl_multi_exec($curl_purges, $running);
-  } while ($running > 0);
+    $mrc = curl_multi_exec($curl_purges, $running);
+  } while ($mrc == CURLM_CALL_MULTI_PERFORM);
+  while ($running && $mrc == CURLM_OK) {
+    if (curl_multi_select($curl_purges) != -1) {
+      do {
+        $mrc = curl_multi_exec($curl_purges, $running);
+      } while ($mrc == CURLM_CALL_MULTI_PERFORM);
+    }
+  }
   ob_end_clean();

   // Result collection. Collects the http code returned for each url purged
