diff --git a/modules/field/views_handler_filter_field_list.inc b/modules/field/views_handler_filter_field_list.inc
index 440d55b..b955e70 100644
--- a/modules/field/views_handler_filter_field_list.inc
+++ b/modules/field/views_handler_filter_field_list.inc
@@ -10,21 +10,7 @@
  *
  * @ingroup views_filter_handlers
  */
-class views_handler_filter_field_list extends views_handler_filter_many_to_one {
-
-  function init(&$view, &$options) {
-    parent::init($view, $options);
-    // Migrate the settings from the old filter_in_operator values to filter_many_to_one.
-    if ($this->options['operator'] == 'in') {
-      $this->options['operator'] = 'or';
-    }
-    if ($this->options['operator'] == 'not in') {
-      $this->options['operator'] = 'not';
-    }
-    $this->operator = $this->options['operator'];
-  }
-
-
+class views_handler_filter_field_list extends views_handler_filter_in_operator {
   function get_value_options() {
     $field = field_info_field($this->definition['field_name']);
     $this->value_options = list_allowed_values($field);
